fix flakey dns tests

from https://travis-ci.org/runconduit/conduit/jobs/349559062:

```
=== RUN   TestInformerUpdate
--- FAIL: TestInformerUpdate (0.00s)
	Error Trace:	dns_test.go:22
			dns.go:137
			dns_test.go:70
	Error:      	Not equal: 
	            	expected: conduit_common.TcpAddress{Ip:(*conduit_common.IPAddress)(0xc42035a030), Port:0x50}
	            	actual  : conduit_common.TcpAddress{Ip:(*conduit_common.IPAddress)(0xc42035a020), Port:0x50}
	            	
	            	Diff:
	            	--- Expected
	            	+++ Actual
	            	@@ -1,2 +1,2 @@
	            	-(conduit_common.TcpAddress) ip:<ipv4:167772161 > port:80 
	            	+(conduit_common.TcpAddress) ip:<ipv4:167772162 > port:80 
	            	 
	Test:       	TestInformerUpdate
	Error Trace:	dns_test.go:22
			dns.go:137
			dns_test.go:70
	Error:      	Not equal: 
	            	expected: conduit_common.TcpAddress{Ip:(*conduit_common.IPAddress)(0xc42035a040), Port:0x50}
	            	actual  : conduit_common.TcpAddress{Ip:(*conduit_common.IPAddress)(0xc42035a010), Port:0x50}
	            	
	            	Diff:
	            	--- Expected
	            	+++ Actual
	            	@@ -1,2 +1,2 @@
	            	-(conduit_common.TcpAddress) ip:<ipv4:167772162 > port:80 
	            	+(conduit_common.TcpAddress) ip:<ipv4:167772161 > port:80 
	            	 
	Test:       	TestInformerUpdate
	Error Trace:	dns_test.go:26
			dns.go:137
			dns_test.go:70
	Error:      	Not equal: 
	            	expected: conduit_common.TcpAddress{Ip:(*conduit_common.IPAddress)(0xc42035a0d0), Port:0x50}
	            	actual  : conduit_common.TcpAddress{Ip:(*conduit_common.IPAddress)(0xc42035a0a0), Port:0x50}
	            	
	            	Diff:
	            	--- Expected
	            	+++ Actual
	            	@@ -1,2 +1,2 @@
	            	-(conduit_common.TcpAddress) ip:<ipv4:167772162 > port:80 
	            	+(conduit_common.TcpAddress) ip:<ipv4:167772163 > port:80 
	            	 
	Test:       	TestInformerUpdate
	Error Trace:	dns_test.go:26
			dns.go:137
			dns_test.go:70
	Error:      	Not equal: 
	            	expected: conduit_common.TcpAddress{Ip:(*conduit_common.IPAddress)(0xc42035a0e0), Port:0x50}
	            	actual  : conduit_common.TcpAddress{Ip:(*conduit_common.IPAddress)(0xc42035a090), Port:0x50}
	            	
	            	Diff:
	            	--- Expected
	            	+++ Actual
	            	@@ -1,2 +1,2 @@
	            	-(conduit_common.TcpAddress) ip:<ipv4:167772163 > port:80 
	            	+(conduit_common.TcpAddress) ip:<ipv4:167772162 > port:80 
	            	 
	Test:       	TestInformerUpdate
```

--- a/controller/destination/dns_test.go
+++ b/controller/destination/dns_test.go
@@ -1,6 +1,7 @@
 package destination
 
 import (
+	"sort"
 	"testing"
 
 	common "github.com/runconduit/conduit/controller/gen/common"
@@ -14,9 +15,24 @@ type testListener struct {
 	removed []common.TcpAddress
 }
 
+// sorting results makes it easier to compare against expected output
+type ByTCPAddr []common.TcpAddress
+
+func (b ByTCPAddr) Len() int           { return len(b) }
+func (b ByTCPAddr) Swap(i, j int)      { b[i], b[j] = b[j], b[i] }
+func (b ByTCPAddr) Less(i, j int) bool { return b[i].String() <= b[j].String() }
+
 func (me testListener) Update(add []common.TcpAddress, remove []common.TcpAddress) {
 	assert.Equal(me.t, len(me.added), len(add))
 	assert.Equal(me.t, len(me.removed), len(remove))
+
+	// Sort lists of TCP addresses before comparison. Ordering is not guaranteed,
+	// as the underlying util.DiffAddresses uses a map to determine changes.
+	sort.Sort(ByTCPAddr(me.added))
+	sort.Sort(ByTCPAddr(add))
+
+	sort.Sort(ByTCPAddr(me.removed))
+	sort.Sort(ByTCPAddr(remove))
 
 	for i, addr := range add {
 		assert.Equal(me.t, me.added[i], addr)

